File: src-tauri/src/models/loaded_page.rs
use super::{Branch, Page, StoredEvent, StoredSnapshot};
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LoadedPage {
    pub page: Page,
    pub branch: Branch,
    pub snapshot: Option<StoredSnapshot>,
    pub events: Vec<StoredEvent>,
}

File: src/domain/schema/loaded-page.ts
import { Schema } from 'effect'
import { Branch } from '~/domain/schema/branch'
import { Page } from '~/domain/schema/page'
import { StoredEvent } from '~/domain/schema/stored-event'
import { StoredSnapshot } from '~/domain/schema/stored-snapshot'

export const LoadedPage = Schema.Struct({
  page: Page,
  branch: Branch,
  snapshot: Schema.NullOr(StoredSnapshot),
  events: Schema.Array(StoredEvent),
})
export type LoadedPage = typeof LoadedPage.Type
